src/discovery/azure/Akka.Discovery.Azure/Actors/AzureDiscoveryGuardian.cs

```diff
@@ -0,0 +1,219 @@
+﻿// -----------------------------------------------------------------------
+//  <copyright file="AzureDiscoveryGuardian.cs" company="Akka.NET Project">
+//      Copyright (C) 2013-2022 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Discovery.Azure.Model;
+using Akka.Event;
+using Akka.Util.Internal;
+
+namespace Akka.Discovery.Azure.Actors
+{
+    internal sealed class AzureDiscoveryGuardian: UntypedActor
```

Comment:
Should add some XML-DOC comments here for future maintainers.

Reply:
Will add in a separate PR